evm/src/main/java/org/hyperledger/besu/evm/EVM.java

```diff
@@ -260,9 +260,20 @@
                   0x9d,
                   0x9e,
                   0x9f -> SwapOperation.staticOperation(frame, opcode - SWAP_BASE);
-              default -> { // unoptimized operations
+                // unoptimized operations
+              default -> {
+                boolean isContextChange = frame.getMessageStackDepth() > 0;
+                if (isContextChange) {
+                  operationTracer.traceContextEnter(frame);
```

Comment:
(a) This will demolish performance for non-context change operations.
(b) it will detect every call off the root later as a context change, not just CALL and CREATE operations.

`org.hyperledger.besu.evm.processor.AbstractMessageProcessor` is the correct place to handle this, likely in the `process` method.  You will still want to check for stack-height zero.